cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -606,6 +606,13 @@ class SnappyUnifiedMemoryManager private[memory](
     }
     memoryForObject.clear()
   }
+
+  // Recovery is a special case. If any of the storage pool has reached 90% of
+  // max storage pool size stop recovery.
+  override def shouldStopRecovery(): Boolean = synchronized {
+    (offHeapStorageMemoryPool.memoryUsed >= (maxOffHeapStorageSize * 0.90) ) ||
+        (onHeapStorageMemoryPool.memoryUsed >= (maxHeapStorageSize * 0.90))
+  }
```

Comment:
use ">" and not ">=" so that for case where there is no off-heap configured, the check will not fail (else for only on-heap case it will fail on very first invocation)